### PR TITLE
TMA-696: Handle status 200 with no content type

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/gooddata/rest/connection.rb [hunk 1]
```diff
@@ -597,6 +596,10 @@ def process_response(options = {}, &block)
         elsif response.code == 204
           result = nil
           GoodData.rest_logger.debug 'Response: 204 no content'
+        elsif response.code == 200 && content_type.nil? && response.body.empty?
+          result = nil
+          # TMA-696
+          GoodData.rest_logger.warn 'Got response status 200 but no content-type and body.'
```
Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Calls 'gooddata.rest_logger' 6 times (DuplicateMethodCall)

File: lib/gooddata/rest/connection.rb [hunk 2]
```diff
@@ -597,6 +596,10 @@ def process_response(options = {}, &block)
         elsif response.code == 204
           result = nil
           GoodData.rest_logger.debug 'Response: 204 no content'
+        elsif response.code == 200 && content_type.nil? && response.body.empty?
```
Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Calls 'response.code' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Refers to 'response' more than self (maybe move it to another class?) (FeatureEnvy)

Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Performs a nil-check (NilCheck)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/554/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/334/ : SUCCESS in 1m 47s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/641/ : SUCCESS in 1m 49s

**Comment by kubamahnert** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/221/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/335/ : SUCCESS in 28s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/642/ : SUCCESS in 1m 44s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1118/ : FAILURE in 42m 49s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/874/ : UNSTABLE in 41m 26s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/223/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/338/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/645/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1120/ : FAILURE in 43m 00s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/876/ : UNSTABLE in 2m 54s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/556/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/340/ : SUCCESS in 1m 41s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/647/ : SUCCESS in 31s

**Comment by kubamahnert** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/225/ : SUCCESS in 24s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/341/ : SUCCESS in 27s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/648/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1122/ : SUCCESS in 47m 08s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/878/ : SUCCESS in 23m 18s

**Pull request merged**